src/data/set/finite.lean

```diff
@@ -72,6 +72,15 @@ theorem exists_finite_iff_finset {p : set α → Prop} :
 ⟨λ ⟨s, hs, hps⟩, ⟨hs.to_finset, hs.coe_to_finset.symm ▸ hps⟩,
   λ ⟨s, hs⟩, ⟨↑s, finite_mem_finset s, hs⟩⟩
 
+lemma finite.fin_param {s : set α} (h : finite s) :
+  ∃ (n : ℕ) (f : fin n → α), injective f ∧ range f = s :=
+begin
+  classical,
+  obtain ⟨f⟩ := nonempty_of_trunc (fintype.equiv_fin (h.to_finset : set α)),
+  exact ⟨_, coe ∘ f.symm, subtype.coe_injective.comp (equiv.injective _),
+    set.ext $ λ x, ⟨λ ⟨y, h⟩, by simpa [ ← h] using (f.symm y).2, λ hs, ⟨f ⟨x, by simpa⟩, by simp⟩⟩⟩
```

Comment:
I think that there's a worthwhile definition to extract here:
```
/-- Given an equivalence to a coerced set `↥s`, produce an embedding to the elements of that set. -/
@[simps]
def set_coe.equiv_as_embedding {α β : Type*} {s : set β} (e : α ≃ ↥s) : α ↪ β :=
⟨coe ∘ e, subtype.coe_injective.comp e.injective⟩

lemma set_coe.equiv_as_embedding_range {α β : Type*} {s : set β} (e : α ≃ ↥s) :
  set.range (set_coe.equiv_as_embedding e) = s :=
set.ext $ λ x, ⟨λ ⟨y, h⟩, h ▸ subtype.coe_prop (e y), λ hs, ⟨e.symm ⟨x, hs⟩, by simp⟩⟩
```

Reply:
I implemented some version of this nice suggestion.